core/templates/dev/head/services/searchServiceSpec.js

```diff
@@ -0,0 +1,182 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests that average ratings are being computed correctly.
+ */
+
+describe('Search service', function() {
```

Comment:
These tests look great, well done!

One thing -- can you check the case where the URL contains ampersands, escaped ampersands, stuff like that? The argument comes in from window.location.search so I'm guessing that the user can manipulate it to be basically anything; it would be good to handle it correctly. We should also update the documentation for searchService.updateSearchFieldsBasedOnUrlQuery() to clarify whether the first argument is expected to be escaped or unescaped -- this is the sort of thing that's really easy to get wrong.